### DISCUSSION TITLE
What does getEtherContractBlindSegmentScript() do in ETHScript.java.

### DISCUSSION THREAD

**Discussion by cyje66** [CoolBitX-Technology/coolwallet-signing-tools#65] [Q&A]
**Description**
We're testing getEtherContractBlindSegmentScript() in ETHScript.java and we're confused. The argument we used: `15328000002c8000003c80000000000000000000000082571ED172c5E202b5309A94aEd0168Bc66F23A70000002386f26fc100000000000000003b9aca0c00000000000000005208000000000000000070a08231`.
But the rawTransaction we got in the script-tester looked a little bit strange, the data field seemed to be incorrect. We looked into the [sdk](https://github.com/CoolBitX-Technology/coolwallet-sdk/blob/ed3fb4c9e7ea10a96241e39033a0133885117d26/packages/coin-eth/src/utils/scriptUtils.ts#L76) to see if we composed the argument incorrectly and we weren't sure about the meaning of the `data`, which is `(handleHex(transaction.data).length / 2).toString(16).padStart(8, '0'); // data length`. If the input data is a function signature(e.g., `70a08231`), doesn't that mean the `data` will always be `00000004`? 

Thus, we changed the data to be `00000004` and the `data` still seems to be incorrect.

**Expected Behavior**

We assume the purpose of this function is to invoke a function call to a smart contract. But the `data` field seemed not to be in this case. Did we miss anything when we compose the argument? Or is there other explanation of the purpose of this function? Please help us here.

**Screenshots or appendix**
* rawTx(we've changed the chainId to 3, input data=function signture)
![image](https://user-images.githubusercontent.com/23631550/156303302-8ed8a9e5-f5e2-4e3d-ad23-fe7913157ea8.png)
* rawTx(input data=`00000004`, the output data=`84`, which is not a correct rlp encoding code)
![image](https://user-images.githubusercontent.com/23631550/156307261-eeedfef0-78e1-4eea-a945-f25e54a37d98.png)

**Comment by hankliu418** [comment]
If you use getEtherContractBlindSegmentScript, the argument is the length of the data. You need to call another command 80A8 to send data, then will get the result you wanted.
We usually use this when the data is larger than 4000 bytes, or we'll use getEtherContractBlindScript.

**Reply by cyje66** [reply]
Where should I input 80A8 command in this picture?
![image](https://user-images.githubusercontent.com/23631550/160998916-40ba6edc-d32c-4ede-9ace-248ba2b1f712.png)

Also, what should I input segment arg, data arg, and script field?
In this case, segment arg is `70a08231`, right?

**Reply by hank121314** [reply]
`segment arg` is a script argument without data length. (script-tester will add data length automatically).
`data arg` is the data field of Ethereum data.
`script` is the script.

**Reply by cyje66** [reply]
Got an error, error code: `609C`, message: `the total length of the inspected data is lower than the signature`

![image](https://user-images.githubusercontent.com/23631550/161007052-625ba8aa-5633-4cf2-b13d-69f3ae4eb93d.png)

`segment arg`: 15328000002c8000003c80000000000000000000000082571ED172c5E202b5309A94aEd0168Bc66F23A70000002386f26fc100000000000000012a05f20000000000000000005208000000000000001c6057361d
`data arg`: the Ethereum data, length > 4000 bytes
`script`: 03050601C707000000003CA00700C2ACD70032FFF8C2ACD7001EFFF6C2ACD70028FFF6CC071094CAA02700C2A2D700FFF6C4ACC7003A04CC0E1004C2E09700CC07C0028080BE0710DC07C003455448D207C005534D415254D207CC05065052455353425554546F4E
